Cli autosuggestions

catreedle opened the pull request:
<!---
Thank you for your pull request! 🚀
-->

### Contributor checklist

<!-- Please replace the empty checkboxes [] below with checked ones [x] accordingly. -->

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
- [x] I have tested my code with the `pytest` command as directed in the [testing section of the contributing guide](https://github.com/scribe-org/Scribe-Data/blob/main/CONTRIBUTING.md#testing)

---

### Description


This PR adds a new file `generate_autosuggestions.py` that extracts steps from  [gen_autosuggestions.ipynb](https://github.com/scribe-org/Scribe-Data/blob/main/src/scribe_data/wikipedia/gen_autosuggestions.ipynb)

Can be tested with command
```bash
scribe-data g -l italian -dt autosuggestions
scribe-data g -l indonesian -dt autosuggestions
```
or from the interactive CLI

### Related issue

<!--- Scribe-Data prefers that pull requests be related to already open issues. -->
<!--- If applicable, please link to the issue by replacing ISSUE_NUMBER with the appropriate number below. -->
<!--- Feel free to delete this section if this does not apply. -->

- #206

axif0 commented:
It takes only 57 minutes and half to download and parse Autosuggestions. :sweat_smile: 
 
![image](https://github.com/user-attachments/assets/d886638e-fd00-41cc-b846-328216977cc8)


Will look into this later, nice work @catreedle   :rocket:

catreedle commented:
> It takes only 57 minutes and half to download and parse Autosuggestions. 😅
> 
> ![image](https://private-user-images.githubusercontent.com/66010854/430609117-d886638e-fd00-41cc-b846-328216977cc8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM4NTAyMjksIm5iZiI6MTc0Mzg0OTkyOSwicGF0aCI6Ii82NjAxMDg1NC80MzA2MDkxMTctZDg4NjYzOGUtZmQwMC00MWNjLWI4NDYtMzI4MjE2OTc3Y2M4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA1VDEwNDUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlN2Q1OWJmODA2MmNkZTJjOTZiOTI2MzFkMjlkZTkxOWMyYzIzZDI3ODFlZmYxOWI0MGU4ZGFiMmVlYzJjZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MMtMqxx6dAVp4Ea3HK2u0m3M4F_QTwhn85pRTxnX0Gg)
> 
> @andrewtavis do we need to add `last-modified` parameter also in autosuggestions?
> 
> ![image](https://private-user-images.githubusercontent.com/66010854/430609524-d4a7b958-8062-4e10-8822-bb93370e0430.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM4NTAyMjksIm5iZiI6MTc0Mzg0OTkyOSwicGF0aCI6Ii82NjAxMDg1NC80MzA2MDk1MjQtZDRhN2I5NTgtODA2Mi00ZTEwLTg4MjItYmI5MzM3MGUwNDMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA1VDEwNDUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlY2U4OGM4MWVjOWIyODUyNzgwM2FmYzdjNmQ2NzEyZjJkOGEyYjBjMGVlZTVmYWE1ZDE4NGU3MjY0ZWRkZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.D4OVYnCAsKyS9liUTB6NNHOQJ6qsFveURFmtB0hMVB8)
> 
> Will look into this later, nice work @catreedle 🚀

oh I set file limit
```python
files = download_wiki(
        language=language,
        target_dir=target_dir,
        file_limit=1,  # Limiting for development purpose
        dump_id=dump_id,
        force_download=force_download,
    )
```

and num words
```python
gen_autosuggestions(
        text_corpus,
        language=language,
        num_words=100,  # Limiting for development purpose
        ignore_words=None,
        update_local_data=True,
        verbose=True,
    )
 ```
 to save time and internet quota. It should be much faster than that. or did you set those off to download the whole dump?

andrewtavis commented:
Nice work to also use the testing arguments like this, @catreedle!

> @andrewtavis do we need to add `last-modified` parameter also in autosuggestions?

I'd honestly say no. Yes we could include it, but this current functionality is kind of a stopgap until next year where the general idea that we've had for a next GSoC project is to integrate language models into the keyboard applications. From there we'd actually have machine learning based autosuggestions based on context and not just single word autosuggestions, so at that point this feature will be redundant. Does this make sense?

I'll a deeper look into this in the coming days. Would be great if the final errors could be resolved, and then I'd try to merge this and the other PR in by say Tuesday evening 😊

andrewtavis commented:
Via the sync right now, @axif0 so you're aware, @catreedle and I discussed that we should remove autosuggestions from the normal get CLI command and have it be separate as the processes are fairly distinct. This will also be included in the PR :)

Thanks, @catreedle!

catreedle commented:
hi @andrewtavis @axif0 
I've added minimum coverage tests and removed autosuggestions from the interactive get prompt.
direct command like `scribe-data g -l swedish -dt autosuggestions` is still available.
I think this is ready for review. looking forward to your feedback. happy to revise if needed 😊

andrewtavis reviewed (approved):
All's looking great here, @catreedle! So wonderful to have your continued support of the project :) This really is a major feature 😊

I noticed that the profanity query isn't working, so I'll make an issue to investigate that :)
